algolia/search/index_interface.go

```diff
@@ -10,7 +10,7 @@ type IndexInterface interface {
 	GetName() string
 	ClearObjects(opts ...interface{}) (res UpdateTaskRes, err error)
 	Delete(opts ...interface{}) (res DeleteTaskRes, err error)
-	Exists() (exists bool, err error)
+	Exists(opts ...interface{}) (exists bool, err error)
```

Comment:
Isn't this a breaking change?

Reply:
No, variadic function can be called with zero or more arguments, so it doesn't change the way we call the function without argument

```
Exists()
Exists(context.Background())
```

Reply:
That's correct... Both calls are compatible but the object implementing the interface will complain. It means we cannot merge it because it will break current implementation. The alternative I can think of are:

- introduce a new method like `ExistsWithOpts` and merge the two in the next major
- reimplement the `Exists()` method on your side and expose the options (the function is small)

Reply:
Ho yes, totally forgot about this ...

Adding a new `ExistsWithOpts` seems a little hacky i think ... We already reimplement the `Exists` function with options on our side 😉 

So maybe you can add this on your todo list for the next major release ?

Reply:
> Adding a new ExistsWithOpts seems a little hacky i think

Yes, it does. 

> We already reimplement the Exists function with options on our side 😉

Perfect.

> So maybe you can add this on your todo list for the next major release ?

Yes. Thanks for your contribution!